internal/expr/parser.go

```diff
@@ -6,15 +6,13 @@ import (
 	"strings"
 )
 
-// A QueryPart represents a section of a parsed SQL statement, which forms
-// a complete query when processed together with its surrounding parts, in
-// their correct order.
-type queryPart interface {
-	// String returns the part's representation for debugging purposes.
+// expression represents a parsed node of the SQLair query's AST.
+type expression interface {
+	// String returns a text representation for debugging and testing purposes.
 	String() string
 
 	// marker method
-	part()
+	expression()
```

Comment:
Keep this one as `expr` to make it distinct from the type name. I hope we can obviate the meed for it in future anyway.

Reply:
Changed